### PR TITLE
fix(medusa): Add inventory decoration for cart endpoints

### LINKED ISSUE #5181: Wrong inventory_quantity using @medusajs/inventory
### Describe the bug

When you are using an inventory module and adding some items to the cart or creating a draft order,
in the cart => items => variant => inventory_quantity has the wrong value (not from the inventory module)

### System information

Medusa version (including plugins):
 "@medusajs/medusa": "^1.14.0",
"@medusajs/inventory": "^1.9.2",
"@medusajs/stock-location": "^1.9.1",

### Steps to reproduce the behavior
1. Create a new product with managed inventory
2. In admin go to product 1, click on "Manage Inventory" on variant a, and enter some stock value, like 12
3. In store, go to product, add it to cart, and check what you will have inside cart => items => variant => inventory_quantity. Your value is probably 0

### Expected behavior

I expect to see `inventory_quantity` 12. If there is some reservation like 2, then `inventory_quantity` should be 10.

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- decorate item totals for `cart.item.variant` when adding a line-item and retrieving a cart

closes #5181

### CODE REVIEW

**Comment by andriinuts** [comment]
@pKorsholm probably there is the same issue in product => variant => inventory_quantity. It displays value not from inventory module.

**Comment by pKorsholm** [comment]
@andriinuts on what endpoint?

**Comment by andriinuts** [comment]
> @andriinuts on what endpoint?

@pKorsholm `/admin/products/{id}` and can't check right now but maybe also in `store` endpoint

**Comment by andriinuts** [comment]
Just checked and in the `store` as result of `medusaClient.products.list` it also not set correct data to the variant `inventory_quantity`

**Comment by pKorsholm** [comment]
This has been a bug on `/store/variants` but was fixed in `1.16.0`. 

have you associated your stock_location with a sales channel?
also do you see this if querying the api without the query client?

**Comment by andriinuts** [comment]
> This has been a bug on `/store/variants` but was fixed in `1.16.0`.
> 
> have you associated your stock_location with a sales channel? also do you see this if querying the api without the query client?

I'll have possibility to check it on `v1.16.0` tomorrow but if you are saying it was fixed in `1.16.0` then I believe it is and if I'll find that something is not working I can create new issue, thx!

**Comment by andriinuts** [comment]
@pKorsholm just checked in v1.16.1 and you are right, it's working as expected in product endpoints in the store. But `/admin/products/{id}` still have this issue

**Comment by andriinuts** [comment]
is this MR blocked?

**Comment by andriinuts** [comment]
@pKorsholm and also I found that it's not working as expected for the search service, data are coming not from the inventory module

**Comment by olivermrbl** [comment]
@pKorsholm, I believe this one is ready to merge. 

> But /admin/products/{id} still have this issue

Will you investigate and get back on this?

File: packages/medusa/src/api/routes/admin/products/get-product.ts [hunk 1]
```diff
@@ -80,9 +92,28 @@ export default async (req, res) => {
 
   const product = rawProduct
 
+  const decoratePromises: Promise<any>[] = []
   if (!shouldSetPricing) {
-    await pricingService.setProductPrices([product])
+    decoratePromises.push(pricingService.setProductPrices([product]))
+  }
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/products/get-product.ts]: comment: Are we not using the flag incorrectly? I would assume the exact opposite behavior i.e. `shouldSetPricing -> setProductPrices`

Comment (pKorsholm) [reply packages/medusa/src/api/routes/admin/products/get-product.ts]: Yes 😂 updated it

**Comment by andriinuts** [comment]
@olivermrbl can you please have a look to close this issue?

**Comment by andriinuts** [comment]
@olivermrbl @adrien2p sory to bother you but please make a review for this fix because it's almost 1 month in review. It's core functionality and it's broken

File: packages/medusa/src/api/routes/store/carts/get-cart.ts [hunk 2]
```diff
@@ -69,7 +76,27 @@ export default async (req, res) => {
...
+    SalesChannelFeatureFlag.key
+  )
+
+  if (salesChannelsEnabled) {
+    select.push("sales_channel_id")
+  }
 
   const data = await cartService.retrieveWithTotals(id, req.retrieveConfig)
+
+  if (shouldSetAvailability && salesChannelsEnabled) {
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/store/carts/get-cart.ts]: question: Do we need to add `&& salesChannelsEnabled` to the store-facing product endpoints?

File: packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts [hunk 3]
```diff
@@ -2,10 +2,16 @@ import { FlagRouter } from "@medusajs/utils"
 import { AwilixContainer } from "awilix"
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts:2]: question: What about when we remove or update line items?

**Review by olivermrbl — APPROVED** [review]
LGTM, a couple of questions :)

**Comment by andriinuts** [comment]
@pKorsholm Hi, will you have time to have a look to the MR this week?

File: packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts (hunk 3 above)
Comment (pKorsholm) [reply packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts:2]: Good point, just went through all cart endpoints and applied this approach where relevant

**Pull request merged**